geopandas/io/file.py

```diff
@@ -27,6 +27,10 @@ def read_file(filename, **kwargs):
             f_filt = f
         gdf = GeoDataFrame.from_features(f_filt, crs=crs)
 
+        # re-order with column order from metadata, with geometry first
+        columns = ["geometry"] + list(f.meta["schema"]["properties"])
```

Comment:
I wasn't aware of an alternative name for the geometry column upon reading from a shapefile. With the current version, I tried writing a `GeoDataFrame` with an alternatively named `geometry` column, and it came back named "geometry". Looking at `geodataframe.GeoDataFrame.from_features`, it seems this name is hardcoded in place.

`read_postgis` seems to place the `geometry` column last, and also retains a different name for the column. So it seems that `geometry` should be the last column, rather than first, as I just placed it.